contracts/sources/obligation.move

```diff
@@ -693,47 +694,103 @@ module suilend::obligation {
+                obligation,
+                borrow,
+                &target_reserve_array_indices,
+                &disabled_pairings_map,
             );
 
-            if (deposit_index < vector::length(&obligation.deposits)) {
+            if (is_borrow_looped) {
                 return true
             };
 
-            // special case for usdc/usdt looping
-            if (borrow.reserve_array_index == usdc_reserve_array_index) {
-                let usdt_deposit_index = find_deposit_index_by_reserve_array_index(
-                    obligation, 
-                    usdt_reserve_array_index
-                );
-                if (usdt_deposit_index < vector::length(&obligation.deposits)) {
-                    return true
-                };
-            };
+            i = i + 1;
+        };
 
-            if (borrow.reserve_array_index == usdt_reserve_array_index) {
-                let usdc_deposit_index = find_deposit_index_by_reserve_array_index(
+        false
+    }
+    
+    // Checks if a given `Borrow` object is looped
+    fun is_borrow_looped<P>(
+        obligation: &Obligation<P>,
+        borrow: &Borrow,
+        target_reserve_array_indices: &vector<u64>,
+        disabled_pairings_map: &vector<vector<u64>>,
+    ): bool {
+        // Check if borrow-deposit reserve match
+        let deposit_index = find_deposit_index_by_reserve_array_index(
+            obligation, 
+            borrow.reserve_array_index
+        );
+
+        if (deposit_index < vector::length(&obligation.deposits)) {
+            return true
+        };
+
+        // Check if it's a target reserve being borrowed
+        let target_borrow_idx = target_borrow_idx(borrow, target_reserve_array_indices);
+
+        // If the borrowing is over a targetted reserve
+        // we check if the deposit reserve is a disabled pair
+        if (option::is_some(&target_borrow_idx)) {
+            let disabled_pairs = vector::borrow(disabled_pairings_map, *option::borrow(&target_borrow_idx));
+            let pair_count = vector::length(disabled_pairs);
+            let i = 0;
+
+            while (i < pair_count) {
+                let disabled_reserve_array_index = *vector::borrow(disabled_pairs, i);
+
+                let deposit_index = find_deposit_index_by_reserve_array_index(
                     obligation, 
-                    usdc_reserve_array_index
+                    disabled_reserve_array_index
                 );
-                if (usdc_deposit_index < vector::length(&obligation.deposits)) {
+                if (deposit_index < vector::length(&obligation.deposits)) {
                     return true
                 };
+            
+                i = i +1;
             };
+        };
 
-            i = i + 1;
+        return false
+    }
+
+    // Checks if it's borrowing a target reserve and if so returns the
+    // index of the such reserve in the `target_reserve_array_indices` vector
+    fun target_borrow_idx(
```

Comment:
i believe we can use vector::index_of() ? if so, i wouldn't abstract this function

Reply:
done